### PR TITLE
NIAD-188 Adds fake-spine support for foward reliable messaging

### PR DESCRIPTION
**Pull request by mattd-kainos**
* Adds support for a interaction using forward-reliable messaging
* Removed unused heads
* Implements using request ConversationId in the reply

### CODE REVIEW

File: integration-tests/fake_spine/fake_spine/vnp_spine_response.py [hunk 1]
```diff
@@ -15,7 +18,8 @@ class VnpSpineResponseException(Exception):
 
 class VnpSpineResponseBuilder(SpineResponseBuilder):
 
-    root_id_expression = re.compile('(<id root="(?P<messageId>.+)"\\s*/>)')
+    root_id_expression = re.compile('<id root="(?P<messageId>.+)"\\s*/>')
```
Comment (michael-kainos) [inline integration-tests/fake_spine/fake_spine/vnp_spine_response.py:21]: is this regex here safe?

File: integration-tests/fake_spine/fake_spine/vnp_responses/async_reliable_inbound.mustache [hunk 2]
```diff
@@ -15,7 +15,7 @@ Content-Type: text/xml
...
...om><eb:To><eb:PartyId eb:type="urn:nhs:names:partyType:ocs+serviceInstance">{{to_party_id}}</eb:PartyId></eb:To><eb:CPAId>S918999404212</eb:CPAId><eb:ConversationId>{{conversation_id}}</eb:ConversationId><eb:Service>urn:nhs:names:services:psis</eb:Service><eb:Action>MCCI_IN010000UK13</eb:Action><eb:MessageData><eb:MessageId>{{message_id}}</eb:MessageId><eb:Timestamp>2020-03-19T14:05:22Z</eb:Timestamp><eb:RefToMessageId>{{ref_to_message_id}}</eb:RefToMessageId></eb:MessageData><eb:DuplicateElimination/></eb:MessageHeader><eb:AckRequested eb:version="2.0" soap:mustUnderstand="1" soap:actor="urn:oasis:names:tc:ebxml-msg:actor:toPartyMSH" eb:signed="false"/><eb:SyncReply eb:version="2.0" soap:mustUnderstand="1" soap:actor="http://schemas.xmlsoap.org/soap/actor/next"/></soap:Header><soap:Body><eb:Manifest xmlns:hl7ebxml="urn:hl7-org:transport/ebXML/DSTUv1.0" eb:version="2.0"><eb:Reference xlink:href="cid:{{message_id}}@spine.nhs.uk"><eb:Schema eb:location="urn:hl7-org:v3_MCCI_IN010000UK13.xsd" eb:version="13"/><eb:Description xml:lang="en">The HL7 payload</eb:Description><hl7ebxml:Payload style="HL7" encoding="XML" version="3.0"/></eb:Reference></eb:Manifest></soap:Body></soap:Envelope>
```
Comment (michael-kainos) [inline integration-tests/fake_spine/fake_spine/vnp_responses/async_reliable_inbound.mustache]: Should we use https for schemas.xmlsoap.org

**Review by michael-kainos — COMMENTED** [review]
Hi @mattd-kainos ,

Please review the comments added

File: integration-tests/fake_spine/fake_spine/vnp_responses/async_reliable_inbound.mustache (hunk 2 above)
Comment (mattd-kainos) [reply integration-tests/fake_spine/fake_spine/vnp_responses/async_reliable_inbound.mustache]: These were scraped from actual opentest replies. I think we should use the same as opentest.

File: integration-tests/fake_spine/fake_spine/vnp_spine_response.py (hunk 1 above)
Comment (mattd-kainos) [reply integration-tests/fake_spine/fake_spine/vnp_spine_response.py:21]: yes, this is an internal testing tool

**Review by michael-kainos — DISMISSED** [review]

**Review by jaroslaw-wojciechowski — DISMISSED** [review]

File: integration-tests/fake_spine/fake_spine/vnp_responses/async_reliable_inbound.mustache (hunk 2 above)
Comment (bartek-sarul) [inline integration-tests/fake_spine/fake_spine/vnp_responses/async_reliable_inbound.mustache]: pretty print please

**Review by bartek-sarul — CHANGES_REQUESTED** [review]

File: integration-tests/fake_spine/fake_spine/vnp_responses/async_reliable_inbound.mustache (hunk 2 above)
Comment (mattd-kainos) [reply integration-tests/fake_spine/fake_spine/vnp_responses/async_reliable_inbound.mustache]: done

**Review by jaroslaw-wojciechowski — APPROVED** [review]

**Review by bartek-sarul — APPROVED** [review]

**Pull request merged**